spec/controllers/users_controller_spec.rb

```diff
@@ -89,22 +89,41 @@
 
   describe '#update' do
     let(:update_params) do
-      { 'contact_name' => 'BAR', 'email' => 'changed@foo.com' }
+      { 'contact_name': 'BAR',
+        'email': 'changed@foo.com' }
     end
+
     context 'when logged in as affiliate' do
       before { activate_authlogic }
       include_context 'approved user logged in'
 
-      it { is_expected.to permit(*permitted_params).for(:update, params: { user: update_params }) }
+      context 'when changing the password' do
+        let(:update_password_params) do
```

Comment:
It would be better to set this variable as `let(:update_params)`. That way you can use `let(:update_user)` to run the update:
```ruby
let(:update_user) do
  post: update, id: current_user.id, user: update_params
end 
```
Then you can DRY up the various `post :update`s with `update_user`, as we discussed.